test/e2e/aws_test.go

```diff
@@ -54,6 +62,24 @@ const (
 	scaleDownReplicas = 0
 )
 
+type statefulSetInfo struct {
+	name                      string
+	namespace                 string
+	replicas                  int32
+	selector                  map[string]string
+	storageClassName          string
+	volumeName                string
+	volumeSize                string
+	svcName                   string
+	svcPort                   int32
+	svcPortName               string
+	containerName             string
+	containerImage            string
+	containerPort             int32
+	podTerminationGracePeriod int64
+	volMountPath              string
+}
+
```

Comment:
Debatable. We can either hardcode magic values while creating the PodTemplateSpec/VolumeClaimTemplate/Service or pass the values as arguments to the functions creating the config objects. To avoid too many arguments to function call and to repetitively use the values created the struct and passed it across function calls.

Reply:
i believe name and namespace are the only fields we would require to pass around and other fields can be hardcoded as they will not be of much use for other test cases.

Reply:
Some of the fields are used it more than one place. Hence, passing them as struct which is readable.